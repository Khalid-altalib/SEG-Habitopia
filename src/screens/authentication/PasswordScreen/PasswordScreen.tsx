import React from "react";
import { View } from "react-native";

import { TextType } from "types";
import Background from "@components/Background";
import PaddedContainer from "@components/PaddedContainer";
import Text from "@components/Text";
import PasswordForm from "@features/auth/PasswordForm/PasswordForm";
import { List } from "native-base";

const PasswordScreen = () => {
  const rules = [
    "Is 8 characters or longer",
    "Contains at least one uppercase and lowercase character",
    "Contains at least one special character",
  ];

  return (
    <Background>
      <PaddedContainer>
        <View style={{ width: "100%", height: "100%", paddingTop: "10%" }}>
          <Text type={TextType.Heading} style={{ marginBottom: 25 }}>
            What's your password?
          </Text>
          <Text type={TextType.Regular} style={{ marginBottom: 5 }}>
            Please ensure password:
          </Text>
          <List marginBottom={4} space={2} borderWidth={0}>
            {rules.map((rule, index) => (
              <Text type={TextType.Subtle} key={index}>
                - {rule}
              </Text>
            ))}
          </List>
          <PasswordForm />
        </View>
      </PaddedContainer>
    </Background>
  );
};

export default PasswordScreen;
