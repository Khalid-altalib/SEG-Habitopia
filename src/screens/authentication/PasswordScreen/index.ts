import PasswordScreen from "./PasswordScreen";

export default PasswordScreen;
