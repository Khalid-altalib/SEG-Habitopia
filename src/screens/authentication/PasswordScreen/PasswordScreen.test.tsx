// React
import React from "react";

// React Native Testing Library
import { render, screen } from "@testing-library/react-native";

// Redux
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

// Habitopia
import PasswordScreen from "@screens/authentication/PasswordScreen";
import TestingWrapper from "@app/testingWrapper";

it("Renders correctly.", () => {
  const mockStore = configureStore([thunk])({});

  render(
    <TestingWrapper store={mockStore}>
      <PasswordScreen />
    </TestingWrapper>
  );

  expect(screen.toJSON()).toMatchSnapshot();
});
