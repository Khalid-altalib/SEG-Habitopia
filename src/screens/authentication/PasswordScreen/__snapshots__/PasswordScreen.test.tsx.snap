// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly. 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      {
        "colors": [
          4293190887,
          4294243573,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      {
        "height": "100%",
      }
    }
  >
    <View
      style={
        {
          "height": "100%",
          "padding": 25,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "height": "100%",
            "paddingTop": "10%",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#18181b",
              },
              {
                "fontFamily": undefined,
                "fontSize": 40,
              },
              {
                "marginBottom": 25,
              },
            ]
          }
        >
          What's your password?
        </Text>
        <Text
          style={
            [
              {
                "color": "#18181b",
              },
              {
                "fontFamily": undefined,
                "fontSize": 19,
              },
              {
                "marginBottom": 5,
              },
            ]
          }
        >
          Please ensure password:
        </Text>
        <View
          dataSet={{}}
          style={
            {
              "borderColor": "#d4d4d8",
              "borderWidth": 0,
              "marginBottom": 16,
              "paddingBottom": 8,
              "paddingTop": 8,
            }
          }
        >
          <Text
            style={
              [
                {
                  "color": "#18181b",
                },
                {
                  "color": "#71717a",
                  "fontFamily": undefined,
                  "fontSize": 15,
                },
                undefined,
              ]
            }
          >
            - 
            Is 8 characters or longer
          </Text>
          <View
            dataSet={{}}
            style={
              {
                "height": 8,
              }
            }
          />
          <Text
            style={
              [
                {
                  "color": "#18181b",
                },
                {
                  "color": "#71717a",
                  "fontFamily": undefined,
                  "fontSize": 15,
                },
                undefined,
              ]
            }
          >
            - 
            Contains at least one uppercase and lowercase character
          </Text>
          <View
            dataSet={{}}
            style={
              {
                "height": 8,
              }
            }
          />
          <Text
            style={
              [
                {
                  "color": "#18181b",
                },
                {
                  "color": "#71717a",
                  "fontFamily": undefined,
                  "fontSize": 15,
                },
                undefined,
              ]
            }
          >
            - 
            Contains at least one special character
          </Text>
        </View>
        <View>
          <Text
            style={
              [
                {
                  "color": "#18181b",
                },
                {
                  "color": "#71717a",
                  "fontFamily": undefined,
                  "fontSize": 15,
                },
                {
                  "marginBottom": 5,
                },
              ]
            }
          >
            Password
          </Text>
          <View
            dataSet={{}}
            isFocused={false}
            style={
              {
                "alignItems": "center",
                "borderColor": "#d4d4d4",
                "borderRadius": 4,
                "borderWidth": 1,
                "flexDirection": "row",
                "overflow": "hidden",
              }
            }
          >
            <TextInput
              accessible={true}
              dataSet={{}}
              editable={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              onKeyPress={[Function]}
              placeholderTextColor="#a3a3a3"
              secureTextEntry={true}
              style={
                {
                  "backgroundColor": "transparent",
                  "color": "#171717",
                  "flex": 1,
                  "fontFamily": undefined,
                  "fontSize": 18,
                  "fontStyle": "normal",
                  "fontWeight": "400",
                  "height": "100%",
                  "paddingBottom": 8,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 8,
                  "width": "100%",
                }
              }
              testID="password-input"
              value=""
            />
          </View>
          <View
            icon={
              <Icon
                allowFontScaling={false}
                color="white"
                name="arrowright"
                size={20}
              />
            }
            isFullWidth={true}
            onPress={[Function]}
            style={
              {
                "marginTop": 20,
              }
            }
            type={0}
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "marginTop": 20,
                  "opacity": 1,
                }
              }
            >
              <ViewManagerAdapter_ExpoLinearGradient
                proxiedProperties={
                  {
                    "colors": [
                      4278220774,
                      4286456526,
                    ],
                    "endPoint": [
                      1,
                      1,
                    ],
                    "locations": undefined,
                    "startPoint": [
                      0,
                      0,
                    ],
                  }
                }
                style={
                  [
                    {
                      "alignItems": "center",
                      "borderRadius": 10,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "paddingHorizontal": 40,
                      "paddingVertical": 15,
                    },
                    {
                      "width": "100%",
                    },
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "color": "#18181b",
                      },
                      {
                        "color": "#e4e4e7",
                        "fontFamily": undefined,
                        "fontSize": 19,
                      },
                      [
                        null,
                        {
                          "alignSelf": "center",
                        },
                      ],
                    ]
                  }
                >
                  Continue
                </Text>
                <View
                  style={
                    {
                      "marginLeft": 5,
                    }
                  }
                >
                  <Text />
                </View>
              </ViewManagerAdapter_ExpoLinearGradient>
            </View>
          </View>
        </View>
      </View>
    </View>
  </ViewManagerAdapter_ExpoLinearGradient>
</RNCSafeAreaProvider>
`;
